### PR TITLE
Clarify instructions in BUILDING.md

### PR DESCRIPTION
**Pull request by offby1**
I'm pretty much a noob when it comes to Java and Android; the things I've fixed here either prevented me from building, or led me to waste effort.

### CODE REVIEW

File: BUILDING.md [hunk 1]
```diff
@@ -19,12 +20,12 @@ The following steps should help you (re)build Signal from the command line.
 2. Make sure you have the [Android SDK](https://developer.android.com/sdk/index.html) installed.
 3. Ensure that the following packages are installed from the Android SDK manager:
     * Android SDK Build Tools
-    * SDK Platform
+    * SDK Platform (API level 22)
```
Comment (odeke-em) [inline BUILDING.md]: This might be misleading as there are multiple API levels that can be used. Is "level 22" the minimum?

File: BUILDING.md [hunk 2]
```diff
@@ -19,12 +20,12 @@ The following steps should help you (re)build Signal from the command line.
...
     * Android Support Repository
     * Google Repository
-4. Create a local.properties file at the root of your source checkout and add an sdk.dir entry to it.
+4. Create a local.properties file at the root of your source checkout and add an sdk.dir entry to it.  For example:
 
-        sdk.dir=\<path to your sdk installation\>
+        sdk.dir=/Application/android-sdk-macosx
```
Comment (odeke-em) [inline BUILDING.md]: Maybe too single platform centric to Mac OSX?

File: BUILDING.md [hunk 3]
```diff
@@ -5,7 +5,8 @@ Basics
 ------
 
 Signal uses [Gradle](http://gradle.org) to build the project and to maintain
-dependencies.
+dependencies.  However, you needn't install it yourself; the
```
Comment (odeke-em) [inline BUILDING.md]: Would you mind taking a look at the Wiki at https://github.com/WhisperSystems/Signal-Android/wiki/How-to-build-Signal-from-the-sources?
If needed maybe make an amend here?

File: BUILDING.md (hunk 1 above)
Comment (offby1) [inline BUILDING.md]: I think it's both the minimum and the maximum -- I no longer remember for sure, but I think there's something in the source that requires that exact version.

File: BUILDING.md (hunk 2 above)
Comment (offby1) [inline BUILDING.md]: Sure, it's specific to OS X, but it's an example; any example will be specific to some OS.  The point is that I didn't realize that I wasn't meant to type the `\<` and `\>` verbatim; I thought an example would be clearer.

**Pull request closed without merging**